### ISSUE TITLE
How to tag a generation?

### ISSUE THREAD

**Issue by TorosFanny** [NixOS/nix#596]
I searched through the document, I didn't find any way to tag a generation. But that would be very useful, since we can't easily remember how a generation differ from others.

**Comment by jb55** [comment]
(triage) is this possible? is it documented anywhere?

**Comment by copumpkin** [comment]
@TorosFanny I don't think it's possible, but it should be. Maybe if you rename the issue to "make it possible" and reopen?

**Comment by TorosFanny** [comment]
Sorry, but too long time passed. I couldn't even understand what I asked.

**Comment by JosephLucas** [comment]
I also think that tagging a generation would be useful.

**Comment by zimbatm** [comment]
I don't think this is strictly necessary. You can either change the profile name, or add the metadata to the derivation output.

**Comment by JosephLucas** [comment]
Ok, I didn't know I could change these file names without disturbing NixOS. I still confuse profile and generation. IIUC a "profile" is a upstream symlink (e.g. `~/.nix-profile`) and "generations" are the symlinks at the level where alternative symlinks exist (e.g. `/nix/var/nix/profiles/per-user/jlucas/profile-97-link`).
What do you mean by changing the profile name ? Do I rename `~/.nix-profile` or `/nix/var/nix/profiles/per-user/<user>/profile` ?

> or add the metadata to the derivation output
How do you do that ?

> I don't think this is strictly necessary.
I agree. But it could be a user-friendly feature.

**Comment by bew** [comment]
I'm still interested

**Comment by jtobin** [comment]
I see that this is closed, but FWIW I'm really surprised this isn't an option (in the spirit of `git-tag`).
